src/Pos/DB/Block.hs

```diff
@@ -102,6 +103,31 @@ loadDataWhile getter predicate start = NewestFirst <$> doIt start
                 then (d :) <$> doIt prev
```

Comment:
That name was already used in the similar function named `loadDataWhile`, I didn't want to change existing code. But, I agree, `go` would be more idiomatic.